sklearn/tests/test_public_functions.py

```diff
@@ -0,0 +1,77 @@
+from importlib import import_module
+from inspect import signature
+
+import pytest
+
+from sklearn.utils._param_validation import generate_invalid_param_val
+from sklearn.utils._param_validation import generate_valid_param
+from sklearn.utils._param_validation import make_constraint
+from sklearn.cluster import kmeans_plusplus
```

Comment:
I am not sure about using the callable directly. We will have to import every single function and it will be difficult once we get quite a lot of functions.

What about accepting the module as a string (e.g. `sklearn.cluster.kmeans_plusplus`) and making a local import in the helper.

Regarding adding functions to the list instead of removing them, I think that I am fine with it and I assume we can change to the second strategy once we have fewer functions to be tested than functions tested.

Reply:
Alternatively, we could import the modules and access to functions via their qualified name (`cluster.kmeans_plusplus`) instead of their full qualified name (`sklearn.cluster.kmeans_plusplus`).